dagshub/data_engine/model/datasource.py

```diff
@@ -1200,6 +1211,73 @@ def contains(self, item: str):
         self._test_not_comparing_other_ds(item)
         return self.add_query_op("contains", item)
 
+    def _periodic_filter(self, item: Union[str, List[str]], periodtype):
+        self._test_not_comparing_other_ds(item)
+        self._query.timezone = _get_local_timezone()
+        return self.add_query_op(periodtype, item)
+
+    def year(self, item: List[str]):
```

Comment:
Additionally: all the list args should be `*args` instead, because I imagine querying for a single year is going to be the most common usecase, and creating a list for that is inconvenient

Reply:
done